### PR TITLE
[5.2] Remove root path only if it is at the beginning

### PR DESCRIPTION
**Pull request by costafrancesco94**
Pull Request for Issue #36669 .

### Summary of Changes
Remove JPATH_ROOT from absolute file path only if JPATH_ROOT is at beginning of string

### Testing Instructions

Added by rdeutz

* Find out the root path of your installation e.g /var/www/joomla
* create a directory /var/www/joomla/media/templates/site/cassiopeia/images/var/www/joomla and copy an image file into the directory e.g. myimage.png
*  go into the template index.php
* add echo HTMLHelper::_('image',  'var/www/joomla/myimage.png', 'theImage', [], true,0); 

### Actual result BEFORE applying this Pull Request
If JPATH_ROOT is repeated inside file names, resources are not getting loaded.

Example:
`JPATH_ROOT = '/joomla'`
`$path = '/joomla/media/system/images/joomla-favicon.svg'`
Wrong output before: `/media/system/images-favicon.svg`

Another less unusual example:
`JPATH_ROOT = '/app/html`
`$path = '/app/html/media/com_something/js/app/html_custom.js'`
Wrong output before: `/media/com_something/js_custom.js`

### Expected result AFTER applying this Pull Request
If `JPATH_ROOT` is present 2 times in the $path, it will be removed only if at beginning (where it should be)

Example:
`JPATH_ROOT = '/joomla'`
`$path = '/joomla/media/system/images/joomla-favicon.svg'`
Correct output with patch: `/media/system/images/joomla-favicon.svg`

Another less unusual example:
`JPATH_ROOT = '/app/html`
`$path = '/app/html/media/com_something/js/app/html_custom.js'`
Wrong output before: `/media/com_something/js/app/html_custom.js`

### Documentation Changes Required

Problem was not declared anywhere else

Code run as before (JPATH_ROOT is not supposed to be in the middle of the $path if not for casuality)

### CODE REVIEW

File: libraries/src/HTML/HTMLHelper.php [hunk 1]
```diff
@@ -1295,7 +1295,13 @@ protected static function addFileToBuffer($path = '', $ext = '', $debugMode = fa
 	 */
 	protected static function convertToRelativePath($path)
 	{
-		$relativeFilePath = Uri::root(true) . str_replace(JPATH_ROOT, '', $path);
+		// Remove JPATH_ROOT only if it is at beginning of full path
+		if (strpos($path, JPATH_ROOT) === 0)
+		{
+			$path = substr_replace($path, '', 0, strlen(JPATH_ROOT));
+		}
+
```
Comment (dgrammatiko) [inline libraries/src/HTML/HTMLHelper.php]:
```suggestion
$path = strpos($path, JPATH_ROOT) === 0 ? substr_replace($path, '', 0, strlen(JPATH_ROOT)) : $path;
```

You can write it as a one line (using the elvis operator)

**Review by dgrammatiko — CHANGES_REQUESTED** [review]

File: libraries/src/HTML/HTMLHelper.php (hunk 1 above)
Comment (costafrancesco94) [reply libraries/src/HTML/HTMLHelper.php]: A lot cooler this way!

File: libraries/src/HTML/HTMLHelper.php [hunk 2]
```diff
@@ -1295,7 +1295,13 @@ protected static function addFileToBuffer($path = '', $ext = '', $debugMode = fa
 	 */
 	protected static function convertToRelativePath($path)
 	{
-		$relativeFilePath = Uri::root(true) . str_replace(JPATH_ROOT, '', $path);
+		// Remove JPATH_ROOT only if it is at beginning of full path
+		if (strpos($path, JPATH_ROOT) === 0)
```
Comment (dgrammatiko) [inline libraries/src/HTML/HTMLHelper.php]: You need to remove this part though:
```js
if (strpos($path, JPATH_ROOT) === 0)
{
   $path = substr_replace($path, '', 0, strlen(JPATH_ROOT));
}
```

Comment (costafrancesco94) [reply libraries/src/HTML/HTMLHelper.php]: Sorry it's the first time I use github PR

Comment (dgrammatiko) [reply libraries/src/HTML/HTMLHelper.php]: You're doing fine

File: libraries/src/HTML/HTMLHelper.php (hunk 1 above)
Comment (richard67) [reply libraries/src/HTML/HTMLHelper.php]: @dgrammatiko I's a normal ternary operator, not Elvis. Elvis would be `?:`.

Comment (dgrammatiko) [reply libraries/src/HTML/HTMLHelper.php]: Yes, that was a ternary...

**Comment by HLeithner** [comment]
This pull request has automatically rebased to 4.2-dev.

**Comment by HLeithner** [comment]
This pull request has been automatically rebased to 4.3-dev.

**Comment by HLeithner** [comment]
This pull request has been automatically rebased to 4.4-dev.

**Comment by HLeithner** [comment]
This pull request has been automatically rebased to 5.2-dev.

File: libraries/src/HTML/HTMLHelper.php [hunk 3]
```diff
@@ -1222,7 +1222,10 @@ protected static function addFileToBuffer($path = '', $ext = '', $debugMode = fa
      */
     protected static function convertToRelativePath($path)
     {
-        $relativeFilePath = Uri::root(true) . str_replace(JPATH_PUBLIC, '', $path);
+        // Remove JPATH_ROOT only if it is at beginning of full path
+        $path = strpos($path, JPATH_ROOT) === 0 ? substr_replace($path, '', 0, \strlen(JPATH_ROOT)) : $path;
```
Comment (joomdonation) [inline libraries/src/HTML/HTMLHelper.php]: The code is not valid anymore because we now use `JPATH_PUBLIC`, not `JPATH_ROOT` as in the past. I think code could be changed to, but no clue how to really test it:

```php
$path = \str_starts_with($path, JPATH_PUBLIC) ? \substr_replace($path, '', 0, \strlen(JPATH_PUBLIC)) : $path;
```

**Comment by fgsw** [comment]
@costafrancesco94 Can the titel be changed to a more meaningful?

File: libraries/src/HTML/HTMLHelper.php [hunk 4]
```diff
@@ -1222,7 +1222,10 @@ protected static function addFileToBuffer($path = '', $ext = '', $debugMode = fa
      */
     protected static function convertToRelativePath($path)
     {
-        $relativeFilePath = Uri::root(true) . str_replace(JPATH_PUBLIC, '', $path);
+        // Remove JPATH_ROOT only if it is at beginning of full path
```
Comment (joomdonation) [inline libraries/src/HTML/HTMLHelper.php]:
```suggestion
        // Remove JPATH_PUBLIC only if it is at beginning of full path
```

File: libraries/src/HTML/HTMLHelper.php (hunk 3 above)
Comment (joomdonation) [inline libraries/src/HTML/HTMLHelper.php]:
```suggestion
        $path = \str_starts_with($path, JPATH_PUBLIC) ? \substr_replace($path, '', 0, \strlen(JPATH_PUBLIC)) : $path;
```

File: libraries/src/HTML/HTMLHelper.php [hunk 5]
```diff
@@ -1222,7 +1222,10 @@ protected static function addFileToBuffer($path = '', $ext = '', $debugMode = fa
      */
     protected static function convertToRelativePath($path)
     {
-        $relativeFilePath = Uri::root(true) . str_replace(JPATH_PUBLIC, '', $path);
+        // Remove JPATH_PUBLIC only if it is at beginning of full path
+        $path = \str_starts_with($path, JPATH_PUBLIC) ? \substr_replace($path, '', 0, \strlen(JPATH_PUBLIC)) : $path;
```
Comment (richard67) [inline libraries/src/HTML/HTMLHelper.php]:
```suggestion
        $path = str_starts_with($path, JPATH_PUBLIC) ? substr_replace($path, '', 0, \strlen(JPATH_PUBLIC)) : $path;
```

**Comment by rdeutz** [comment]
Added testimng instructions and tested successfully

**Comment by rdeutz** [comment]
I have tested this item :white_check_mark: successfully on d47f76c798623072847d63c0817543a8c6e74798<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36685">issues.joomla.org/tracker/joomla-cms/36685</a>.</sub>

**Comment by joomdonation** [comment]
I have tested this item :white_check_mark: successfully on d47f76c798623072847d63c0817543a8c6e74798<br /><br />On windows, it's hard to follow the instructions. But I reviewed code, also tried to using media manager, navigate to different pages and do not see any error.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36685">issues.joomla.org/tracker/joomla-cms/36685</a>.</sub>

**Comment by joomdonation** [comment]
RTC. Thanks.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36685">issues.joomla.org/tracker/joomla-cms/36685</a>.</sub>

**Comment by richard67** [comment]
@michaelmaass Hmm, it doesn't make much sense to test a pull request which was merged long time ago and so is included since Joomla 5.2. Are you sure you marked the right one?
